lib/collection/src/shard/local_shard.rs

```diff
@@ -155,14 +157,19 @@ impl LocalShard {
                 });
                 continue;
             }
-            let segment = match load_segment(&segments_path) {
-                Ok(x) => x,
-                Err(err) => panic!(
-                    "Can't load segments from {}, error: {}",
-                    segments_path.to_str().unwrap(),
-                    err
-                ),
-            };
+            load_handlers.push(thread::spawn(move || load_segment(&segments_path)));
```

Comment:
What is the upper bound on the number of concurrent segments here? Just wondering about the thread usage.

Am alternative would be to turning it into an async. blocking task to let Tokio manage the blocking thread pool.

Reply:
theoretically, there are no upper bound, but I would expect that in real life it won't exceed 200. Also the performance will be limited by IO rather than CPU

Reply:
I would prefer to keep it simple, since this operation is only performed during the start-up